### PR TITLE
Bug fix hparam logging with metrics

### LINKED ISSUE #1228: Add support to log hparams and metrics to tensorboard?
How can I log metrics (_e.g._ validation loss of best epoch) together with the set of hyperparameters?

I have looked through the docs and through the code.
It seems like an obvious thing, so maybe I'm just not getting it.

Currently, the only way that I found was to extend the logger class:
```python
class MyTensorBoardLogger(TensorBoardLogger):

    def __init__(self, *args, **kwargs):
        super(MyTensorBoardLogger, self).__init__(*args, **kwargs)

    def log_hyperparams(self, *args, **kwargs):
        pass

    @rank_zero_only
    def log_hyperparams_metrics(self, params: dict, metrics: dict) -> None:
        params = self._convert_params(params)
        exp, ssi, sei = hparams(params, metrics)
        writer = self.experiment._get_file_writer()
        writer.add_summary(exp)
        writer.add_summary(ssi)
        writer.add_summary(sei)
        # some alternative should be added
        self.tags.update(params)
```
And then I'm writing the hparams with metrics in a callback:
```
def on_train_end(self, trainer, module):
        module.logger.log_hyperparams_metrics(module.hparams, {'val_loss': self.best_val_loss})
```
But that doesn't seem right. 
Is there a better way to write some metric together with the hparams as well?

#### Environment

 - OS: Ubuntu18.04
 - conda4.8.3
 - pytorch-lightning==0.7.1
- torch==1.4.0

### PR DESCRIPTION
**Pull request by justusschock**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CHANGELOG.md)?

## What does this PR do?
Fixes #1228 and follows #1630 

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by festeh** [comment]
I remember I did exactly inverse PR in #777. The problem was that for each single run it created two records in the Tensorboard (because new SummaryWriter was created for hparams) and runs view was really cluttered. I wonder if this PR will reintroduce this behavoir.

**Comment by justusschock** [comment]
Okay, I think I already know a solution to this one :) Thanks for the hint @festeh

File: pytorch_lightning/loggers/tensorboard.py [hunk 1]
```diff
@@ -115,13 +115,8 @@ def log_hyperparams(self, params: Union[Dict[str, Any], Namespace],
             )
         else:
             from torch.utils.tensorboard.summary import hparams
```
Comment (olineumann) [inline pytorch_lightning/loggers/tensorboard.py:117]:
```suggestion
```

Isn't needed anymore right?

Comment (olineumann) [reply pytorch_lightning/loggers/tensorboard.py:117]: It crashes if sanitized_params or metrics is None. Tensorboard doc says it is okay, but code checks for dict or raises an error.

[Doc](https://pytorch.org/docs/stable/tensorboard.html)
[Code](https://pytorch.org/docs/stable/_modules/torch/utils/tensorboard/writer.html#SummaryWriter.add_hparams)

File: pytorch_lightning/loggers/tensorboard.py [hunk 2]
```diff
@@ -115,13 +115,8 @@ def log_hyperparams(self, params: Union[Dict[str, Any], Namespace],
...
-            if metrics is None:
-                metrics = {}
-            exp, ssi, sei = hparams(sanitized_params, metrics)
-            writer = self.experiment._get_file_writer()
-            writer.add_summary(exp)
-            writer.add_summary(ssi)
-            writer.add_summary(sei)
+
+            self.experiment.add_hparams(sanitized_params, metrics)
```
Comment (olineumann) [inline pytorch_lightning/loggers/tensorboard.py:119]:
```suggestion
            if sanitized_params is not None and metrics is not None:
                self.experiment.add_hparams(sanitized_params, metrics)
```

Comment (olineumann) [reply pytorch_lightning/loggers/tensorboard.py:119]: In the next version of pytorch hparam or metric equals None removed from doc and also crashes currently. 
https://pytorch.org/docs/master/tensorboard.html

**Comment by Borda** [comment]
@justusschock how about the suggestions from @olineumann?

File: pytorch_lightning/loggers/tensorboard.py (hunk 2 above)
Comment (justusschock) [reply pytorch_lightning/loggers/tensorboard.py:119]: But this would be a breaking change because you can't log hparams without metrics taht way :)

**Comment by justusschock** [comment]
Reverted some of the changes and added logging for metrics if provided

**Comment by justusschock** [comment]
Only the formatting test fails and this one is not related to this PR. I think this should be ready to merge @Borda @williamFalcon

**Comment by Borda** [comment]
I have been fixing formatting in #1786 it is related to `flake8` check all errors now
